app/src/main/java/app/revanced/integrations/returnyoutubedislike/ReturnYouTubeDislike.java

```diff
@@ -163,67 +197,109 @@ public static void onComponentCreated(Object conversionContext, AtomicReference<
                 return;
             }
 
-            // Have to block the current thread until fetching is done
-            // There's no known way to edit the text after creation yet
-            RYDVoteData votingData;
-            long fetchStartTime = 0;
-            try {
-                Future<RYDVoteData> fetchFuture = getVoteFetchFuture();
-                if (fetchFuture == null) {
-                    LogHelper.printDebug(() -> "fetch future not available (user enabled RYD while video was playing?)");
-                    return;
-                }
-                if (SettingsEnum.DEBUG.getBoolean() && !fetchFuture.isDone()) {
-                    fetchStartTime = System.currentTimeMillis();
-                }
-                votingData = fetchFuture.get(MILLISECONDS_TO_BLOCK_UI_WHILE_WAITING_FOR_FETCH_VOTES_TO_COMPLETE, TimeUnit.MILLISECONDS);
-            } catch (TimeoutException e) {
-                LogHelper.printDebug(() -> "UI timed out waiting for fetch votes to complete");
-                return;
-            } finally {
-                recordTimeUISpentWaitingForNetworkCall(fetchStartTime);
-            }
-            if (votingData == null) {
-                LogHelper.printDebug(() -> "Cannot add dislike to UI (RYD data not available)");
-                return;
+            Spanned replacement = waitForFetchAndUpdateReplacementSpan((Spanned) textRef.get(), isSegmentedButton);
+            if (replacement != null) {
+                textRef.set(replacement);
             }
+        } catch (Exception ex) {
+            LogHelper.printException(() -> "Error while trying to update dislikes", ex);
+        }
+    }
 
-            if (updateDislike(textRef, isSegmentedButton, votingData)) {
-                LogHelper.printDebug(() -> "Updated dislike span to: " + textRef.get());
-            } else {
-                LogHelper.printDebug(() -> "Ignoring already updated dislike span: " + textRef.get());
+    public static Spanned onShortsComponentCreated(Spanned span) {
```

Comment:
Unlike onComponentCreated, this code does not need to check, if `conversionContextString(...)` contains the required span. How is this the case & can this be reflected to `conversionContextString(...)` as well? The parameter  `conversionContext` from the method `conversionContextString(...)`  actually is an object of which the `StringBuilder` field can be read from. This way the entire objects `toString()` override does not need to be called and instead the relevant string can be checked for the component name.